compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -4079,6 +4086,7 @@ private ParserRuleContext getNextRuleForEqualOp() {
             case LET_EXPR_LET_VAR_DECL:
             case LET_CLAUSE_LET_VAR_DECL:
             case ENUM_MEMBER_LIST:
+            case FUNC_DEF_OR_FUNC_TYPE:
                 return ParserRuleContext.EXPRESSION;
```

Comment:
need to switch the context to `VAR_DECL_STMT` and remove the change in L2488

Reply:
Here, It can be both a function definition or function type. If it is a function definition we can't change context to `VAR_DECL_STMT`.

Reply:
Since we r inside `getNextRuleForEqualOp()` we know it is a `VAR_DECL_STMT` there after

Reply:
Fixed.